File: Decorator/PHP/HouseBlend.php
<?php
  class HouseBlend extends Beverage {
    public function __construct() {
      parent::__construct();
      $this->description = " HouseBlend";
    }

    public function cost() {
      return 8.9;
    }

    public function getDescription() {
      return $this->description;
    }
  }

File: Decorator/Go/decorator/houseblend.go
package decorator

type HouseBlend struct {
  *Coffee
  price float64
  description string
}

func NewHouseBlend() *HouseBlend {
  return &HouseBlend{&Coffee{}, 3.0, "HouseBlend"}
}

func (self *HouseBlend) Cost() float64 {
  return self.price + self.Coffee.Cost()
}

func (self *HouseBlend) GetDescription() string {
  return self.description + self.Coffee.GetDescription()
}
